app/src/test/java/com/x8bit/bitwarden/data/vault/datasource/sdk/VaultSdkSourceTest.kt

```diff
@@ -1047,58 +1039,34 @@ class VaultSdkSourceTest {
         }
 
     @Test
-    fun `registerFido2Credential should invoke checkUser when called by the SDK`() = runTest {
-        val checkUserResult = CheckUserResult(true, true)
+    fun `registerFido2Credential should return Failure when BitwardenException is thrown`() =
+        runTest {
+            val mockCipherView = createMockCipherView(1)
 
-        val checkUserOptionsSlot = slot<CheckUserOptions>()
-        val uiHintSlot = slot<UiHint>()
-        val mockUserInterface = mockk<Fido2UserInterface> {
             coEvery {
-                checkUser(
-                    capture(checkUserOptionsSlot),
-                    capture(uiHintSlot),
+                fido2.register(
+                    any(),
+                    any(),
+                    any(),
                 )
-            } returns checkUserResult
-        }
+            } throws BitwardenException.E("mockException")
 
-        val mockCipherView = createMockCipherView(number = 1)
-        val mockAttestation = mockk<PublicKeyCredentialAuthenticatorAttestationResponse>()
-        val mockCheckUserOptions = CheckUserOptions(true, Verification.REQUIRED)
-        coEvery { fido2.register(any(), any(), any()) } coAnswers {
-            mockUserInterface.checkUser(
-                mockCheckUserOptions,
-                UiHint.InformNoCredentialsFound,
-            )
-            mockAttestation
-        }
-        vaultSdkSource.registerFido2Credential(
-            request = RegisterFido2CredentialRequest(
-                userId = "mockUserId",
-                origin = "www.bitwarden.com",
-                requestJson = "requestJson",
-                clientData = ClientData.DefaultWithCustomHash(
-                    hash = DEFAULT_SIGNATURE.toByteArray(),
+            val result = vaultSdkSource.registerFido2Credential(
+                RegisterFido2CredentialRequest(
```

Comment:
Thoughts on extracting this to a reusable `const val` to avoid repeating it in each test?